jupyterhub/services/auth.py

```diff
@@ -854,37 +904,32 @@ async def _token_for_code(self, code):
 
         return token_reply['access_token']
 
-    def _encode_state(self, state):
-        """Encode a state dict as url-safe base64"""
-        # trim trailing `=` because = is itself not url-safe!
-        json_state = json.dumps(state)
-        return (
-            base64.urlsafe_b64encode(json_state.encode('utf8'))
-            .decode('ascii')
-            .rstrip('=')
-        )
+    # state-related
 
-    def _decode_state(self, b64_state):
-        """Decode a base64 state
+    oauth_state_max_age = Integer(
+        600,
+        config=True,
+        help="""Max age (seconds) of oauth state.
+        
+        Governs both oauth state cookie Max-Age,
+        as well as the in-memory _oauth_states cache.
+        """,
+    )
+    _oauth_states = Instance(
+        _ExpiringDict,
+        allow_none=False,
+        help="""
+        Store oauth state info, such as next_url
```

Comment:
Should we have something like `Store oauth state info per service` to make things clearer ?

Reply:
added more detail here